mesheryctl/internal/cli/root/mesh/deploy.go

```diff
@@ -16,12 +17,22 @@ import (
 	"github.com/spf13/viper"
 )
 
+// checkArgs checks whether the user has supplied an adapter(-a) argument
+func checkArgs(n int) cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if len(args) < n || args[0] == "" {
```

Comment:
should the check be for min arguments of length 2, i.e "-a" and "adapter-name" ?

Reply:
"-a" and "--adapter-name" are the same thing, so one of them has to be supplied. The command hangs at "Verifying pre-requisites" no arguments are supplied. But works with other valid arguments that are not "-a". So the number of arguments cannot be zero and the error message is intended to tell the user which argument is required.

Reply:
@Zanda256 I meant to ask, if user provides "-a" option and no valid adapter-name for e.g. meshery-linkerd, this checkArgs function would not alert the user, right ?
`mesheryctl mesh deploy -a`